### ISSUE TITLE
UnconditionalIfStatement: False negative when boolean literal is wrapped in UnaryExpression

### ISSUE THREAD

**Issue by leemeii** [pmd/pmd#6435]
**Affects PMD Version:**  7.20.0

**Rule:** UnconditionalIfStatement
<https://pmd.github.io/pmd/pmd_rules_java_errorprone.html#unconditionalifstatement>

**Description:**
UnconditionalIfStatement fails to detect cases where the condition is an effectively constant boolean expression, such as a UnaryExpression involving a boolean literal (e.g., if (!false)).

In the AST, !false is parsed as a UnaryExpression containing a BooleanLiteral, which does not satisfy the current [BooleanLiteral[1]] predicate. This allows unconditional code blocks to bypass static analysis **simply** by adding a logical NOT operator.

**Code Sample demonstrating the issue:**

```java
public class Bug {
    public void method() {
        
        if (!false) { //<-should report (FN)
            int i = 1;
        }
    }
}
```

**Expected outcome:**

PMD should report a violation at line 4, but doesn't. This is a false-negative. The rule should be updated to account for simple constant expressions or at least unary-negated literals.

**Actual outcome**
Line 4 is ignored.

**Running PMD through:** *CLI*

**Comment by adangel** [comment]
Thanks for reporting.

> bypass static analysis simply by adding a logical NOT operator.

By adding single not, you would change the semantic. But the easier way to bypass this rule is `!!true` ...
